sycl/include/CL/sycl/detail/spirv.hpp

```diff
@@ -133,21 +133,22 @@ EnableIfNativeBroadcast<T, IdT> GroupBroadcast(T x, IdT local_id) {
   using GroupIdT = typename GroupId<Group>::type;
   GroupIdT GroupLocalId = static_cast<GroupIdT>(local_id);
   using OCLT = detail::ConvertToOpenCLType_t<T>;
+  using WidenedT = conditional_t<
+      std::is_same<OCLT, cl_char>() || std::is_same<OCLT, cl_short>(), cl_int,
+      conditional_t<
+          std::is_same<OCLT, cl_uchar>() || std::is_same<OCLT, cl_ushort>(),
+          cl_uint,
+          conditional_t<std::is_same<OCLT, _Float16>::value, cl_float, OCLT>>>;
```

Comment:
Implicit conversion between floating-point types will likely alter the precision. We should probably bitcast floating-points to integer to have a simple `sext/zext` + `trunc` pair not affecting any significant bits of the value.

Reply:
`half` is now handled by the bitcast overload, so it is first bitcast to `int16_t` (and then widened to `int32_t`.